### DISCUSSION TITLE
Customizing k3s startup

### DISCUSSION THREAD

**Discussion by dozer75** [rancher-sandbox/rancher-desktop#6728] [Q&A] [answered]
Is there any way to pass additional arguments to k3s in Rancher Desktop? I would like to pass some extra arguments using `kube-apiserver-arg` ...

**Comment by jandubois** [comment, marked as answer]
You have to use a provisioning script for this. Search for `K3S_EXEC` on https://docs.rancherdesktop.io/how-to-guides/provisioning-scripts/#macos--linux

**Reply by dozer75** [reply]
Thanks for answer!

The example shows how to do it on macOS/Linux, however Windows has other files and unfortunately the examples is a bit different, so the question is how do we do the same with provisioning on Windows with e.g. K3s? Would be nice to have the same examples in all environments.. 

But I'll investigate this further!

**Reply by jandubois** [reply]
See https://github.com/rancher-sandbox/rancher-desktop/issues/4595 for a way to set this in a Windows provisioning script.

Make sure the script doesn't have Windows `\r\n` line endings, but just `\n` Unix style line endings!

**Reply by dozer75** [reply]
Thanks!
